### ISSUE TITLE
Not work on Yosemite OSX 10.10

### ISSUE THREAD

**Issue by mems** [mschrag/speedlimit#13]
[`ipfw`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/Manpages/man8/ipfw.8.html) has been depreciated since OSX 10.7 and removed in OSX 10.10. It has been replaced by [`pfctl`](https://developer.apple.com/library/mac/documentation/Darwin/Reference/Manpages/man8/pfctl.8.html#//apple_ref/doc/man/8/pfctl)

More infos:
http://wayback.archive.org/web/20121218001958/http://blog.segment7.net/2009/07/27/bandwidth-limiting-with-pf-and-altq

**Comment by adityapunjani** [comment]
+1

**Comment by ghost** [comment]
+1

**Comment by robertoschwald** [comment]
+1

**Comment by mtshare** [comment]
+1

**Comment by seanxiaoxiao** [comment]
+1

**Comment by maddox** [comment]
Apple provides the exact same functionality via the Network Link Conditioner. You can download it via the [Apple Developer Downloads](https://developer.apple.com/downloads/index.action?q=Network%20Link%20Conditioner). There's really no reason for this project still be active (which is why it isn't)

More info on the Network Link Conditioner can be [found here](http://nshipster.com/network-link-conditioner/).

**Comment by ghost** [comment]
Chrome [dev tools](https://developer.chrome.com/devtools/docs/device-mode#network-conditions) also has this feature.

**Comment by lacek** [comment]
@maddox Network Link Conditioner is not working on localhost, speedlimit does.

**Comment by denji** [comment]
Network Link Conditione official XCode http://nshipster.com/network-link-conditioner/

![network-link-conditioner-system-preference](https://cloud.githubusercontent.com/assets/3759759/5898560/4fc8326c-a559-11e4-9bfc-c561445aded9.png)

![network-link-conditioner-preset](https://cloud.githubusercontent.com/assets/3759759/5898564/59501818-a559-11e4-8cf5-31627a77b937.png)

**Comment by denji** [comment]
Dropped support system traffic prioritization and bandwidth control ALTQ which OpenBSD 5.6 changes since last release came built-in packet filter new queue management system network packets. Remove ALTQ due to the difficulty of maintenance of two systems;

http://www.openbsd.org/cgi-bin/man.cgi/OpenBSD-current/man5/pf.conf.5?query=pf%2econf&sec=5#x5155455545494e47

Packets can be assigned to queues for the purpose of bandwidth control. At least one declaration is required to configure queues, and later any packet filtering rule can reference the defined queues by name. During the filtering component of pf.conf, the last referenced queue name is where any passed packets will be queued, while for blocked packets it specifies where any resulting ICMP or TCP RST packets should be queued. If the referenced queue does not exist on the outgoing interface the default queue for that interface is used. Queues attached to an interface build a tree, thus each queue can have further child queues. Only leaf queues, i.e. queues without children, can be used to assign packets to. The root queue must specifically reference an interface, all other queues pick up the interface(s) they should be created on from their parent queues unless explicitly specified.
In the following example, a queue named std is created on the interface em0, with 3 child queues ssh, mail and http.

**Comment by maddox** [comment]
> @maddox Network Link Conditioner is not working on localhost, speedlimit does.

I've only ever used this locally. I mean, this was made specifically as a dev tool, of course it works locally.

@lacek did you try using `127.0.0.1`?

**Comment by lacek** [comment]
@maddox I tried both `127.0.0.1` and `localhost`, neither works. Tested on Chrome and Firefox with `100% Loss` Profile, once the conditioner's on I cannot access GitHub while `127.0.0.1` and `localhost` are still accessible instantly.

**Comment by ImanMh** [comment]
@maddox Why would you think there is no reason for continuing this project? is Network Link Conditioner open source?

**Comment by Glutexo** [comment]
Apple’s Network Link Conditioner doesn’t work for me on `localhost` either. Using `127.0.0.1` instead does not help. I hoped that speedlimit might be a solution, but since it is discontinued, it apparently isn’t. :(

**Comment by ibc** [comment]
@ghost said:

> Chrome dev tools also has this feature.

Well, unless you need to limit not just HTTP stuff.

**Comment by jlongman** [comment]
See https://github.com/jlongman/speedlimit.

- No binary at the moment however.  
- ~~Haven't tested localhost, if~~ it works ~~with pf it should work.~~

**Comment by jlongman** [comment]
ah here: https://github.com/jlongman/speedlimit/releases - I'm new to doing releases though.

**Comment by denji** [comment]
Hardware IO Tools for Xcode (`Network Link Conditioner`) should work too macOS/iOS.

**Comment by jlongman** [comment]
Yes and it's more configurable (at least in the UI) but apparently it has problems with localhost which speedlimiter does not. I've not tested localhost with NLC myself, I have and it does work with speedlimiter.
